Source/src/modules/ModuleResources.cpp

```diff
@@ -333,6 +339,30 @@ bool Hachiko::ModuleResources::ValidateAssetResources(const YAML::Node& meta) co
     return true;
 }
 
+void Hachiko::ModuleResources::ClearUnusedResources(const std::set<UID>& seen_uids)
+{
+    const auto& library_paths = preferences->GetLibraryPathsMap();
+    for (auto it = library_paths.begin(); it != library_paths.end(); ++it)
+    {
+        const PathNode folder = FileSystem::GetAllFiles(it->second.c_str());
+        ClearLibrary(folder, seen_uids);
+    }
+    HE_LOG("Library cleanup finished.");
+}
+
+void Hachiko::ModuleResources::ClearLibrary(const PathNode& folder, const std::set<UID>& seen_uids)
+{
+    // Iterate all files found in assets except metas and scene
+    for (const PathNode& path_node : folder.children)
```

Comment:
We cannot use the is file check because resources do not have extension and it is not detected as a file. but since all library files are at the same level this is not an issue if we do it this way.